test/functional/api/vim_spec.lua

```diff
@@ -571,6 +571,28 @@ describe('API', function()
       eq({0,7,1,0}, funcs.getpos('.'))
       eq(false, nvim('get_option', 'paste'))
     end)
+    it('Replace-mode', function()
+      -- Within single line
+      nvim('put', {'line 1', 'line 2', 'line 3'}, "c", true, false)
+      command('normal 2l')
+      command('startreplace')
+      nvim('paste', 'new', true, -1)
+      expect([[
+      linew1
```

Comment:
This example is a bit degenerate as two out of three chars are the same, maybe use `normal l` instead of `2l`.

Reply:
Agreed. Will update the test soon™